core/src/main/java/org/commonjava/maven/ext/core/impl/NexusStagingMavenPluginRemovalManipulator.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2012 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.maven.ext.core.impl;
+
+import org.commonjava.maven.ext.common.model.Project;
+import org.commonjava.maven.ext.core.ManipulationSession;
+import org.commonjava.maven.ext.core.state.NexusStagingMavenPluginRemovalState;
+import org.commonjava.maven.ext.core.state.PluginState;
+import org.commonjava.maven.ext.core.state.State;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Named;
+import javax.inject.Singleton;
+
+import java.util.List;
+import java.util.Set;
+
+/**
+ * {@link Manipulator} implementation that can remove nexus-staging-maven-plugin from a project's pom file.
+ * Configuration is stored in a {@link PluginState} instance, which is in turn stored in the
+ * {@link ManipulationSession}.
+ */
+@Named("nexus-staging-maven-plugin-removal-manipulator")
+@Singleton
+public class NexusStagingMavenPluginRemovalManipulator
+        extends PluginRemovalManipulator {
+    private static final Logger logger = LoggerFactory.getLogger( NexusStagingMavenPluginRemovalManipulator.class );
+
+    private ManipulationSession session;
+
+    /**
+     * Initialize the {@link PluginState} state holder in the {@link ManipulationSession}. This state holder detects
+     * version-change configuration from the Maven user properties (-D properties from the CLI) and makes it available for
+     * later.
+     *
+     * @param session
+     */
+    @Override
+    public void init( ManipulationSession session )
+    {
+        super.init( session );
```

Comment:
@rnc Is `init` really supposed to be called from the constructor? This sets the state to the one in the superclass at first, which I don't really want. Because `init` comes from a common interface already. I assumed that it would already be called for us.

Reply:
init is normally called from ManipulationManager. So I'm not really following your comment?

Reply:
Sorry, I almost feel like I should delete that comment as it didn't make sense. I make reference to a constructor which isn't there.

In order to extend `PluginRemovalManipulator`, I had to add a default constructor and call `super.init()` so it can get a reference to the session. But, it also sets the state, too, which I don't necessarily want.

In any case, should I instead move the shared code to say, a common abstract base class and have them extend that one? The reason for extending it is I am using its code directly to make the changes.

Reply:
My vote would be to remove the ` this.session = session;` as its redundant and leave it as is.

Reply:
I agree its not the nicest code having to call `super.init`. Perhaps add `@NoArgsConstructor` to the parent class and then you can remove the super call?

Reply:
I removed `super.init()` call. I am not sure that @NoArgsConstructor is necessary as one is added by default and I didn't seem to need it.

Reply:
Ok cool.